### PR TITLE
chore(ts-migration): creates refine generic type

### PR DESCRIPTION
**Pull request by nunomaduro**
This pull request makes the refine type easier to reuse.

### CODE REVIEW

File: src/connectors/autocomplete/connectAutocomplete.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ export type AutocompleteRendererOptions = {
   /**
    * Searches into the indices with the provided query.
    */
-  refine: (query: string) => void;
+  refine: Refine<string>;
```
Comment (Haroenv) [inline src/connectors/autocomplete/connectAutocomplete.ts:54]: the name "query" is now gone from autocomplete, is that correct?

```
refine(
    ^^^query: string
```

Comment (nunomaduro) [reply src/connectors/autocomplete/connectAutocomplete.ts:54]: Correct.

Comment (Haroenv) [reply src/connectors/autocomplete/connectAutocomplete.ts:54]: Tried to find a solution but couldn't :( https://www.typescriptlang.org/play/index.html#code/C4TwDgpgBAZgdgHgCoDkCGBbCA+KBeKACgGcB7LAcQjggCcBLAY3SwC4pjgG4BzASny4AbqXoATAFASA9NKjAAFvWJRlcAOTAoQtABtxMuaEixEqTDnxEA2uawBddp279B20ZInHo9OJzRwjNAE8AjqGCAAShBgumhBWHDALBDq2FKMpH5aAB7svv6BwUQR0bHxEInJFgJ4uACucGIQML4QnjmE6gCSUADuygryCtBo9cCkmRixEMDQmfW6YlA89ELQpTFxCdTVWKrZEGjLpDAc5BBUNAzMFup8UrJQnQ9AA

Comment (nunomaduro) [reply src/connectors/autocomplete/connectAutocomplete.ts:54]: I think it's fine, 👍

**Comment by nunomaduro** [comment]
Got it!

**Comment by francoischalifour** [comment]
@Haroenv Can you explain your POV? I think this type is completely valid, we use it everywhere. If this is because of the argument name not displaying in IntelliSense, we've got the type _and_ the documentation now.

**Comment by Haroenv** [comment]
the argument is called `query`, `page`, `value`, which I find more useful names than `value` in all cases. I didn't suggest we should immediately close this PR, just giving my pov in a way that prevented the PR to immediately be merged.

At the end of the day the extracted type didn't seem to bring that much value (to me), since it's a function which takes one argument & returns void, it's ok if that's repeated in every connector (for me)

**Pull request closed without merging**